### PR TITLE
[SDK] Encrypted entity client with downloading

### LINKED ISSUE #6955: Task: EntityClient with Encryption in Rust (downloading)
<p><strong>AC</strong></p>
<ul>
    <li>There is an <code>EntityClient</code> that supports encryption and that is
        distinct from the regular <code>EntityClient</code></li>
    <li>Entity client supports downloading and decrypting single ETs and LETs</li>
    <li>Key resolution and encryption are hidden from the interface</li>
</ul>

**Tasks**
 - [ ] Implement decryption of entities given type model, encrypted data and session key

<p><strong>Notes</strong></p>
<ul>
    <li>First version doesn't need encryption, only decryption</li>
   <li>Compression can be omitted in this version</li>
    <li>This version can omit updating <code>ownerEncSessionKey</code></li>
    <li>We do not need to implement permissions</li>
    <li><code>CryptoEntityClient</code> depends on &quot;regular&quot;
        entity client
    </li>
    <li><code>CryptoFacade</code>or it's alternative can also depend on
        &quot;regular&quot; <code>EntityClient</code> if it needs to download
        something
    </li>
    <li>Ultimately depends on <em>Key resolution</em> task but can be written in
        parallel
    </li>
    <li>Depends on #6953</li>
    <li>Depends on #6956</li>
<li> Depends on #7062 </li>
<li> Depends on #7070 </li>
    <li>The model we need to implement is described in <a
            href="https://next.tutao.de/doc/mail-body-migration.html#_simplification_of_the_permission_system"
            class="external-link" rel="nofollow">doc</a> under &quot;Target model&quot;
    </li>
    <li>Separate EntityClient with/without encryption will help us to avoid
        dependency cycles
    </li>
</ul>

### PR DESCRIPTION
**Pull request by rezbyte**
This PR introduces 'CryptoEntityClient' a wrapper around `EntityClient` that utilizes `CryptoFacade` to provide automatic encryption/decryption to entity operations. Only the entity downloading functionality has been implemented as per the issue.

Closes #6955.

### CODE REVIEW

File: tuta-sdk/rust/src/crypto_entity_client.rs [hunk 1]
```diff
@@ -0,0 +1,158 @@
+use std::sync::Arc;
```
Comment (rezbyte) [inline tuta-sdk/rust/src/crypto_entity_client.rs:1]: TODO: Change this to return typed entities.

File: tuta-sdk/rust/src/crypto_entity_client.rs [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+    ) -> Result<T, ApiCallError> {
+        let type_ref = T::type_ref();
+        let type_model = self.entity_client.get_type_model(&type_ref)?;
+        let mut parsed_entity = self.entity_client.load(&type_ref, id).await?;
+
+        if type_model.encrypted {
+            let possible_session_key = self.crypto_facade
+                .resolve_session_key(&mut parsed_entity, type_model)
+                .map_err(
+                    |error| ApiCallError::InternalSdkError {
```
Comment (charlag) [inline tuta-sdk/rust/src/crypto_entity_client.rs:44]: I am not sure if we need a more specific error about key resolution at some but it we can probably keep it like it is for now

**Review by charlag — APPROVED** [review]
looks good! Please squash a little bit

File: tuta-sdk/rust/src/crypto_entity_client.rs (hunk 2 above)
Comment (rezbyte) [reply tuta-sdk/rust/src/crypto_entity_client.rs]: That is true, we should have more detail in the error. I have pushed a new commit now.

**Pull request merged**